### PR TITLE
Add username

### PR DESCRIPTION
**Pull request by Palakkgoyal**
This PR closes issue no. #574 
Done all the required changes such as checking whether the user exist or not. If not then, create a new user doc in users collection and if does exists then, check whether they have a username or not. If not then, give them uid as their username and if exists then, do not do anything in both sign in and sign up page.

### CODE REVIEW

File: src/pages/Login/index.jsx [hunk 1]
```diff
@@ -93,12 +91,25 @@ const LoginScreen = () => {
...
           await usernameDoc.doc(auth.currentUser.uid).set({
             uid: val.user.uid,
+            username: val.user.uid,
             name: val.user.displayName,
             photoURL: val.user.photoURL,
             displayName: val.user.displayName,
             Friends: [],
             posts: [],
           });
+        } else {
```
Comment (narayan954) [inline src/pages/Login/index.jsx:101]: I was thinking, if we make sure to add usernames in signup only we won't need to query for username and give it a default everytime if it's a old sign up as this is a costly operation, what do you say?

**Comment by narayan954** [comment]
Hi @Palakkgoyal can you please resolve conflicts? If you face issues, let me know.

File: src/pages/Login/index.jsx (hunk 1 above)
Comment (Palakkgoyal) [reply src/pages/Login/index.jsx:101]: But, what if the user is new and they tried to sign up through login page. I mean they are new and they click on the google sign in.

**Comment by Palakkgoyal** [comment]
> Hi @Palakkgoyal can you please resolve conflicts? If you face issues, let me know.

Sure.

**Comment by Palakkgoyal** [comment]
I solved the merge issue. Can you please check.

**Comment by narayan954** [comment]
> I solved the merge issue. Can you please check.

new pr? or should I reopen this??

**Comment by Palakkgoyal** [comment]
Ohk. Thanks a lot.

File: src/pages/Login/index.jsx (hunk 1 above)
Comment (narayan954) [reply src/pages/Login/index.jsx]: I meant, in the else condition where existing users sign in, we're querying if username is set or not... and that won't be needed as they already had signed up with username

Comment (narayan954) [reply src/pages/Login/index.jsx]: alright, I'm letting this fix persist and will remove it after clearing users without usernames

**Review by narayan954 — APPROVED** [review]
looks good to me

**Comment by narayan954** [comment]
Thanks a lot for fixing this!

**Comment by Palakkgoyal** [comment]
No worries.

**Pull request merged**